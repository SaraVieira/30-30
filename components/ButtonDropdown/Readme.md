
```example
<ButtonDropdown buttonContent="Components">
  <ul>
    <li><a href="http://goggle.com">Button</a></li>
    <li>Button Dropwndown</li>
    <li>Button Dropdown</li>
    <li>Checkbox</li>
    <li>Weather Widget</li>
    <li>Camera Component</li>
  </ul>
</ButtonDropdown>
```
