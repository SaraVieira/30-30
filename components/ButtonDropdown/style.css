@import '../css/variables.css';

.ButtonDropdownWrapper {
  position: relative;
  display: inline-flex;
  color: var(--white-two);
  align-items: center;
  border-radius: 3px;

  & .ButtonDropdownIcon {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 16px 18px;
    border-left: solid 1px var(--white-two);
    cursor: pointer;
    outline: none;
    border: none;
    background-color: var(--dark-sky-blue-three);
    color: var(--white-two);
    margin-left: -2px;
    min-height: 49px;
    border-radius: 0 3px 3px 0;

    &:hover {
      background-color: color(var(--dark-sky-blue-three) l(+10%));
    }
  }

  & .dropdown {
    position: absolute;
    top: 100%;
    border-radius: 3px;
    background-color: var(--white);
    width: 100%;
    padding: 10px;
    color: var(--dark-sky-blue-three);

    & a {
      color: var(--dark-sky-blue-three);
      text-decoration: none;

      &:hover {
        color: color(var(--dark-sky-blue-three) b(+10%));
      }
    }

    & ul {
      list-style: none;
      padding: 0;
      margin: 0;

      & li:not(:last-child) {
        padding-bottom: 10px;
      }
    }
  }
}
