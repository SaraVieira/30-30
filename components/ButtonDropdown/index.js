// @flow
import React, { Component } from 'react';
import Button from '../Button';
import style from './style.css';

type Props = {
  buttonContent: string,
  children: Element
}

type State = {
  opened: boolean
}

class ButtonDropdown extends Component {
  props: Props;

  constructor(props: Props) {
    super(props);
    this.state = {
      opened: false
    };
    this.handleDrowpdown = this.handleDrowpdown.bind(this);
  }

  state: State;
  handleDrowpdown: () => void;

  handleDrowpdown() {
    this.setState({
      opened: !this.state.opened
    });
  }


  render() {
    return (
      <div className={style.ButtonDropdownWrapper}>
        <Button
          onClick={this.handleDrowpdown}
        >
          {this.props.buttonContent}
        </Button>
        <button
          onClick={this.handleDrowpdown}
          className={style.ButtonDropdownIcon}
        >
          { this.state.opened ? '\u25B2' : '\u25BC' }
        </button>
        { this.state.opened &&
          <div className={style.dropdown}>{this.props.children}</div>
        }
      </div>
    );
  }
}

export default ButtonDropdown;
