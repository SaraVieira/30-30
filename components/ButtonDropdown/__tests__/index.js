import React from 'react'; // eslint-disable-line
import tjson from 'enzyme-to-json';
import { mount } from 'enzyme';
import ButtonDropdown from '../index';

test('ButtonDropdown renders the text inside it', () => {
  const buttonDropdown = mount(
    <ButtonDropdown>Test</ButtonDropdown>
  );
  expect(buttonDropdown.hasClass('ButtonDropdownWrapper')).toEqual(true);
  expect(buttonDropdown.text()).toBe('▼');
  const tree = tjson(buttonDropdown);
  expect(tree).toMatchSnapshot();
});

test('ButtonDropdown renders children on click', () => {
  const buttonDropdown = mount(
    <ButtonDropdown>Test</ButtonDropdown>
  );
  buttonDropdown.find('.primary').simulate('click');
  expect(buttonDropdown.hasClass('ButtonDropdownWrapper')).toEqual(true);
  expect(buttonDropdown.text()).toBe('▲Test');
  const tree = tjson(buttonDropdown);
  expect(tree).toMatchSnapshot();
});

test('ButtonDropdown renders children on click', () => {
  const buttonDropdown = mount(
    <ButtonDropdown>
      <p>Test</p>
    </ButtonDropdown>
  );
  buttonDropdown.find('.primary').simulate('click');
  expect(buttonDropdown.hasClass('ButtonDropdownWrapper')).toEqual(true);
  expect(buttonDropdown.find('p').html()).toBe('<p>Test</p>');
  const tree = tjson(buttonDropdown);
  expect(tree).toMatchSnapshot();
});
