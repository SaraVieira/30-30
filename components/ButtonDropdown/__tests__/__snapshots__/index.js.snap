// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDropdown renders children on click 1`] = `
<ButtonDropdown>
  <div
    className="ButtonDropdownWrapper"
  >
    <Button
      onClick={[Function]}
    >
      <button
        className="button primary"
        onClick={[Function]}
      />
    </Button>
    <button
      className="ButtonDropdownIcon"
      onClick={[Function]}
    >
      ▲
    </button>
    <div
      className="dropdown"
    >
      Test
    </div>
  </div>
</ButtonDropdown>
`;

exports[`ButtonDropdown renders children on click 2`] = `
<ButtonDropdown>
  <div
    className="ButtonDropdownWrapper"
  >
    <Button
      onClick={[Function]}
    >
      <button
        className="button primary"
        onClick={[Function]}
      />
    </Button>
    <button
      className="ButtonDropdownIcon"
      onClick={[Function]}
    >
      ▲
    </button>
    <div
      className="dropdown"
    >
      <p>
        Test
      </p>
    </div>
  </div>
</ButtonDropdown>
`;

exports[`ButtonDropdown renders the text inside it 1`] = `
<ButtonDropdown>
  <div
    className="ButtonDropdownWrapper"
  >
    <Button
      onClick={[Function]}
    >
      <button
        className="button primary"
        onClick={[Function]}
      />
    </Button>
    <button
      className="ButtonDropdownIcon"
      onClick={[Function]}
    >
      ▼
    </button>
  </div>
</ButtonDropdown>
`;
